client/lib/purchases/assembler.js

```diff
@@ -80,26 +76,25 @@ function createPurchaseObject( purchase ) {
 	};
 
 	if ( 'credit_card' === purchase.payment_type ) {
-		const payment = Object.assign( {}, object.payment, {
+		const payment = {
+			...object.payment,
 			creditCard: {
 				id: Number( purchase.payment_card_id ),
 				type: purchase.payment_card_type,
 				processor: purchase.payment_card_processor,
 				number: purchase.payment_details,
 				expiryDate: purchase.payment_expiry,
-				expiryMoment: purchase.payment_expiry
-					? i18n.moment( purchase.payment_expiry, 'MM/YY' )
-					: null,
+				expiryDateValid: Boolean( purchase.payment_expiry ),
+				expiryDateFormat: 'MM/YY',
```

Comment:
`expiryDateFormat` is not a credit-card-specific data, it's just a constant. Then why is it on the `creditCard` record?

And why is `expiryDateFormat` set to `null` on certain objects?

Reply:
As I understand, there are two places where expiry dates appear: in the payment data, and in the credit card data. In the credit card data, they're always in the format of `MM/YY`. In the payment data, they can be either be in the default date format, or in `MM/YY` format in the case of PayPal Direct.

To try to make things consistent, I opted to add an `expiryDateFormat` field to both of them. It defaults to null (no specific format defined, i.e., just use the default date parsing), with `MM/YY` instead where applicable. When parsing with `moment`, you can simply do `moment( expiryDate, expiryDateFormat )` and have it parse the date according to the right format, or do something similar for other libraries.

This seemed to me like an extensible solution that makes it easy to change logic in the future if needed, without offloading that logic onto consumers.

Reply:
It's true! If I look at a purchase I made with credits, the `expiry_date` is in ISO format. For credit card and PayPal, I didn't verify that, but it seems that `expiry_date` is `12/20`.

Reply:
I came here to comment on the same thing 😉 

I wonder if it would make sense to make a decision and go with a single format in all cases. These moment objects that use that expiration field, after all, are being additionally formatted upon rendering of the date. Could using a single format be a good way to remove the unnecessary complexity here? Does it have any downsides?

Reply:
Looking at the PHP code a bit closer, I think we can get rid of the special formatting strings.

The purchase itself can have three timestamps:
- `expiry_date`
- `subscribed_date`
- `renew_date`
All of them are either empty or always ISO strings. Some purchases never expire, e.g., if you buy a premium theme, it's yours forever.

Then there is a separate timestamp, `purchase_expiry_date` (transformed by the Calypso data layer to `purchase.payment.expiryDate`, which is present only on credit card and PayPal purchases, and this one is always in the `MM/YY` format.

Reply:
I think I finally fully grokked this. There's actually three (possible) expiry dates in the Calypso data layer object. There's `purchase.expiryDate`, which is always empty/ISO string; there's `purchase.payment.expiryDate`, which is only set in the case of PayPal Direct and is always `MM/YY`; and there's `purchase.payment.creditCard.expiryDate`, which is only set in the case of credit card and is always `MM/YY`.

The code has been updated to remove `expiryDateFormat` in light of these fixed formats.